domain-models-runtime/src/main/java/org/folio/rest/persist/cache/CachedConnectionManager.java

```diff
@@ -0,0 +1,111 @@
+package org.folio.rest.persist.cache;
+
+import io.vertx.core.Vertx;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Pool;
+import io.vertx.core.Future;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.utils.Envs;
+
+import java.util.Optional;
+
+public class CachedConnectionManager {
+  private static final Logger LOG = LogManager.getLogger(CachedConnectionManager.class);
+  private final ConnectionCache connectionCache = new ConnectionCache();
+  private final int maxPoolSize;
+  private int connectionReleaseDelaySeconds = -1;
+
+  public CachedConnectionManager() {
+    this.maxPoolSize = getMaxSharedPoolSize();
+  }
+
+  public int getCacheSize() {
+    return connectionCache.size();
+  }
+
+  public void clearCache() {
+    connectionCache.clear();
+    LOG.debug("Cleared connection manager");
+  }
+
+  public void removeFromCache(CachedPgConnection connection)  {
+    connectionCache.remove(connection);
+  }
+
+  public void tryAddToCache(CachedPgConnection connection) {
+    connectionCache.tryAdd(connection);
+    this.tryRemoveOldestAvailableConnectionAndClose();
+  }
+
+  public Future<PgConnection> getConnection(Vertx vertx, Pool pool, String schemaName, String tenantId) {
+    if (! PostgresClient.isSharedPool()) {
+      LOG.debug("Not in shared pool mode");
+      return pool.getConnection().map(PgConnection.class::cast);
+    }
+
+    if (this.connectionReleaseDelaySeconds == -1) {
+      throw new IllegalArgumentException("Connection release delay has not been set");
+    }
+
+    LOG.debug("In shared pool mode");
+    return getOrCreateCachedConnection(vertx, pool, schemaName, tenantId);
+  }
+
+  public void setConnectionReleaseDelay(int seconds) {
+    this.connectionReleaseDelaySeconds = seconds;
+  }
+
+  private static int getMaxSharedPoolSize() {
```

Comment:
It looks like this method is only used in the constructor of this class and it is static because the object has not been created yet. Consider moving this into a utilities class or similar. There should be a function in RMB somewhere to derive this configuration.

Reply:
Gonna statically define this.